docs/developer-guide/working-with-plugins.md

```diff
@@ -18,6 +18,21 @@ module.exports = {
 };
 ```
 
+### Preprocessors in Plugins
+
+You can also create plugins that would tell ESLint how to process files other then JavaScript. In order to create a
+preprocessor, object that is exported from your module has to conform to the following interface:
+
+```js
+{
+    preprocess: function(string) { return [string] } //Takes text of the file and returns array of texts to be processed
+    postprocess: function([[Message]]) { return [Message] //Takes array of array of error messages, one for each text
```

Comment:
Oh I see, you're totally right. Thanks for explaining.

Here's what I envision:

```js
// whole plugin
module.exports = {

    rules: {},
    rulesConfig: {},
    processors: {
        ".md": {
            preprocess: function() {},
            postprocess: function() {}
        }
    }
};
```

Would that work?